### ISSUE TITLE
fake socket does not support method send

### ISSUE THREAD

**Issue by sk-** [gabrielfalcao/HTTPretty#251]
Trying to use `aiohttp`(`asyncio`) with `HTTPretty` fails because fake socket does not support the method send.
Note: To get to this state I needed to first fix issue #250.

``` python
Traceback (most recent call last):
  File "test.py", line 15, in wrapper
    loop.run_until_complete(future)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/base_events.py", line 316, in run_until_complete
    return future.result()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/futures.py", line 275, in result
    raise self._exception
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/tasks.py", line 234, in _step
    result = coro.throw(exc)
  File "test.py", line 48, in test_aiohttp
    r = yield from aiohttp.request('put', url, data='Hello World!')
  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/client.py", line 104, in request
    read_until_eof=read_until_eof)
  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/client.py", line 182, in request
    yield from resp.start(conn, read_until_eof)
  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/client.py", line 811, in start
    self.message = yield from httpstream.read()
  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/streams.py", line 501, in read
    result = yield from super().read()
  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/streams.py", line 364, in read
    yield from self._waiter
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/futures.py", line 386, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/tasks.py", line 287, in _wakeup
    value = future.result()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/futures.py", line 275, in result
    raise self._exception
  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/client.py", line 695, in write_bytes
    yield from ret
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/streams.py", line 301, in drain
    raise exc
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/selector_events.py", line 700, in write
    n = self._sock.send(data)
  File "venvs/asyncio/lib/python3.4/site-packages/httpretty/core.py", line 444, in send
    return self.debug('send', *args, **kwargs)
  File "venvs/asyncio/lib/python3.4/site-packages/httpretty/core.py", line 435, in debug
    raise RuntimeError("\n".join(message))
RuntimeError: HTTPretty intercepted and unexpected socket method call.
Please open an issue at 'https://github.com/gabrielfalcao/HTTPretty/issues'
And paste the following traceback:

[b'  File "test.py", line 56, in <module>\n    unittest.main()\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/main.py", line 93, in __init__\n    self.runTests()\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/main.py", line 244, in runTests\n    self.result = testRunner.run(self.test)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/runner.py", line 168, in run\n    test(result)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/suite.py", line 84, in __call__\n    return self.run(*args, **kwds)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/suite.py", line 122, in run\n    test(result)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/suite.py", line 84, in __call__\n    return self.run(*args, **kwds)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/suite.py", line 122, in run\n    test(result)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/case.py", line 625, in __call__\n    return self.run(*args, **kwds)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/case.py", line 577, in run\n    testMethod()\n', b'  File "test.py", line 15, in wrapper\n    loop.run_until_complete(future)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/base_events.py", line 304, in run_until_complete\n    self.run_forever()\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/base_events.py", line 276, in run_forever\n    self._run_once()\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/base_events.py", line 1172, in _run_once\n    handle._run()\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/events.py", line 120, in _run\n    self._callback(*self._args)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/tasks.py", line 292, in _wakeup\n    self._step(value, None)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/tasks.py", line 238, in _step\n    result = next(coro)\n', b'  File "test.py", line 48, in test_aiohttp\n    r = yield from aiohttp.request(\'put\', url, data=\'Hello World!\')\n', b'  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/client.py", line 104, in request\n    read_until_eof=read_until_eof)\n', b'  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/client.py", line 180, in request\n    resp = req.send(conn.writer, conn.reader)\n', b'  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/client.py", line 724, in send\n    request.send_headers()\n', b'  File "venvs/asyncio/lib/python3.4/site-packages/aiohttp/protocol.py", line 654, in send_headers\n    self.transport.write(headers)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/streams.py", line 272, in write\n    self._transport.write(data)\n', b'  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/asyncio/selector_events.py", line 700, in write\n    n = self._sock.send(data)\n', b'  File "venvs/asyncio/lib/python3.4/site-packages/httpretty/core.py", line 444, in send\n    return self.debug(\'send\', *args, **kwargs)\n', b'  File "venvs/asyncio/lib/python3.4/site-packages/httpretty/core.py", line 426, in debug\n    lines = list(map(utf8, traceback.format_stack(frame)))\n']
```

**Comment by mindflayer** [comment]
I am the author of Mocket, which does support asyncio.
https://github.com/mindflayer/python-mocket/

**Comment by gabrielfalcao** [comment]
closed by f0bb2bdc1f5e2771e0cb854f57014e7d0fcc6b71
